src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -650,6 +662,55 @@ public Collection<NodeProvisioner.PlannedNode> provision(
         return Collections.emptyList();
     }
 
+    private static void checkKubernetesUrlFIPS(String url) {
+        if (!FIPS140.useCompliantAlgorithms() || StringUtils.isBlank(url)) {
+            return;
+        }
+        if (!url.startsWith("https:")) {
+            throw new IllegalArgumentException(Messages.KubernetesCloud_kubernetesServerUrlIsNotSecure());
+        }
+    }
+
+    private static void checkSkipTlsVerify(boolean skipTlsVerify) {
+        if (FIPS140.useCompliantAlgorithms() && skipTlsVerify) {
+            throw new IllegalArgumentException(Messages.KubernetesCloud_skipTlsVerifyNotAllowedInFIPSMode());
+        }
+    }
+
+    private static void checkServerCertificateFIPS(String serverCertificate) {
+        if (!FIPS140.useCompliantAlgorithms()) {
+            return;
+        }
+        if (StringUtils.isBlank(serverCertificate)) {
+            throw new IllegalArgumentException(Messages.KubernetesCloud_serverCertificateKeyEmpty());
+        }
+        try {
+            PEMEncodable pem = PEMEncodable.decode(serverCertificate);
```

Comment:
NB if you only have a certificate and do not want to rely on BOuncyCastle you may be able to use

```java
CertificateFactory f = CertificateFactory.getInstance("X.509");

try (InputStream is = new ByteArrayInputStream(serverCertificate.getBytes(StandardCharsets.UTF_8))) {
  Certificate cert = f.generateCertificate(is);
  // the checks....
} catch (CertificateException | AsserttionError /* BC FIPS */  e) {
  // generally invalid...
}

Reply:
> given it was an optional dependency?

it's not optional, right? `bouncycastle-api` was test dependency and now it's a regular dependency. Or is there other possible bouncycastle dependency?

Reply:
Oh! you mean that before this PR the dependency wasn't there, right?

Reply:
https://github.com/jenkinsci/kubernetes-plugin/pull/1590#discussion_r1684306399

Reply:
ah! but that's outdated, so this comment doesn't apply, do it?

Reply:
the comment applies if the author does not want to include a dependency on boucycastle).  I do not see a problem with including a dependency on bouncycastle  for `PEMEncodable` in this PR.

Reply:
It's only used in these lines, I'll move it to plain java so we can leave the pom as it was, no need to add a dependency if we can checkit by other means